handshake/src/prelude.rs

```diff
@@ -0,0 +1,171 @@
+//! Contains macros for use in this crate
```

Comment:
This is a copy and paste of `bitcoin/src/prelude.rs`, realizing I should just import it from the `bitcoin` crate.

Reply:
instead of importing, it might be better to move anything really useful to core

Reply:
I see that macros now live in `bitcoins/src/macros.rs`. I need to refactor the `handshake` crate to follow this pattern. Also agree that the macros could live in the `coins-core` crate. That seems like it would be a better PR on its own.